geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -10479,53 +10527,70 @@ public boolean isInternalRegion() {
     return this.cacheServiceProfiles.getSnapshot();
   }
 
-  public void addCacheServiceProfile(CacheServiceProfile profile) {
-    this.cacheServiceProfiles.put(profile.getId(), profile);
+  @Override
+  public void addCacheServiceProfile(CacheServiceProfile profile)
+      throws IncompatibleCacheServiceProfileException {
+    synchronized (this.cacheServiceProfiles) {
+      CacheServiceProfile cacheServiceProfile =
+          this.cacheServiceProfiles.putIfAbsent(profile.getId(), profile);
+      if (cacheServiceProfile == null) {
+        this.validateProfiles();
+      } else {
+        throw new IncompatibleCacheServiceProfileException(
+            "A profile for: " + profile.getId() + " has already been defined");
+      }
+
+    }
+  }
+
+  @Override
+  public void removeCacheServiceProfile(CacheServiceProfile profile) {
+    synchronized (this.cacheServiceProfiles) {
+      this.cacheServiceProfiles.remove(profile.getId(), profile);
+    }
   }
 
+  protected void validateProfiles() throws IncompatibleCacheServiceProfileException {}
```

Comment:
This is a no op?  So is the call to validateProfiles actually calling of the new machinery?